src/components/Edge-Conn-PCIe.stanza

```diff
@@ -0,0 +1,943 @@
+        pin pcie.data.lane[11].TX.N at Point(-18.000, -40.640) with :
+          direction = Left
+          length = 3.810
+          number-size = 1.270
+          name-size = 1.270
+        pin pcie.data.lane[12].TX.P at Point(-18.000, -43.180) with :
+          direction = Left
+          length = 3.810
+          number-size = 1.270
+          name-size = 1.270
+        pin pcie.data.lane[12].TX.N at Point(-18.000, -45.720) with :
+          direction = Left
+          length = 3.810
+          number-size = 1.270
+          name-size = 1.270
+        pin pcie.data.lane[13].TX.P at Point(-18.000, -48.260) with :
+          direction = Left
+          length = 3.810
+          number-size = 1.270
+          name-size = 1.270
+        pin pcie.data.lane[13].TX.N at Point(-18.000, -50.800) with :
+          direction = Left
+          length = 3.810
+          number-size = 1.270
+          name-size = 1.270
+        pin pcie.data.lane[14].TX.P at Point(-18.000, -53.340) with :
+          direction = Left
+          length = 3.810
+          number-size = 1.270
+          name-size = 1.270
+        pin pcie.data.lane[14].TX.N at Point(-18.000, -55.880) with :
+          direction = Left
+          length = 3.810
+          number-size = 1.270
+          name-size = 1.270
+        pin pcie.data.lane[15].TX.P at Point(-18.000, -58.420) with :
+          direction = Left
+          length = 3.810
+          number-size = 1.270
+          name-size = 1.270
+        pin pcie.data.lane[15].TX.N at Point(-18.000, -60.960) with :
+          direction = Left
+          length = 3.810
+          number-size = 1.270
+          name-size = 1.270
+
+
+
+
+  draw("foreground") = Text("ki_keywords: pcie", 1.27, C, loc(0.0, 0.0), "", TrueTypeFont, true, false)
+  draw("foreground") = Text("ki_fp_filters: *PCIexpress*", 1.27, C, loc(0.0, 0.0), "", TrueTypeFont, true, false)
+  draw("foreground") = Text(">REF", 1.27, W, loc(2.194, -67.310))
+  draw("foreground") = Text(">VALUE", 1.27, W, loc(2.194, -69.850))
+  draw("foreground") = Text("Footprint: Connector_PCBEdge:BUS_PCIexpress", 1.27, C, loc(0.0, 22.860), "", TrueTypeFont, true, false)
+  draw("foreground") = Text("Datasheet: http://www.ritrontek.com/uploadfile/2016/1026/20161026105231124.pdf#page=63", 1.27, C, loc(-1.270, -6.350), "", TrueTypeFont, true, false)
+  draw("foreground") = Text("Description: PCI Express bus connector", 1.27, C, loc(0.0, 0.0), "", TrueTypeFont, true, false)
+  draw("background") = Rectangle(36.000, 129.540)
+
+public pcb-component component (xwidth : jsl/protocols/pcie/PCIeWidth) :
+  description = "PCI Express parameterized edge connector"
+  mpn = "Edge_PCI_Express"
+  datasheet = "http://www.ritrontek.com/uploadfile/2016/1026/20161026105231124.pdf#page=63"
+  reference-prefix = "J"
+
+  if xwidth == PCIe-x32 :
+    fatal("Unsupported width: %_", % [xwidth])
+  val width = jsl/protocols/pcie/PCIe-enum-to-int(xwidth)
+  port pcie : pcie(width, PCIe-PRSNT#)
```

Comment:
[The PR I have up](https://github.com/JITx-Inc/jsl/pull/234) changes this interface a bit. The better interface is `pcie-with-hotplug(width)` from there. 

I can submit a follow up PR to connectors when that gets merged.

Reply:
Cool.